apis/v1alpha2/gateway_types.go

```diff
@@ -397,17 +397,18 @@ const (
 
 	// This reason is used with the "ResolvedRefs" condition when the
 	// Listener has a TLS configuration with at least one TLS CertificateRef
-	// that is invalid or cannot be resolved.
+	// that is malformed or does not exist.
 	ListenerReasonInvalidCertificateRef ListenerConditionReason = "InvalidCertificateRef"
 
 	// This reason is used with the "ResolvedRefs" condition when an invalid or
 	// unsupported Route kind is specified by the Listener.
 	ListenerReasonInvalidRouteKinds ListenerConditionReason = "InvalidRouteKinds"
 
 	// This reason is used with the "ResolvedRefs" condition when
-	// one of the Listener's Routes has a BackendRef to an object in
-	// another namespace, where the object in the other namespace does
-	// not have a ReferenceGrant explicitly allowing the reference.
+	// one of the Listener's Routes references an object in another namespace,
+	// where the object in the other namespace does not have a ReferenceGrant
+	// explicitly allowing the reference. The unpermitted reference can be
+	// either a BackendRef, or a CertificateRef.
```

Comment:
A couple thoughts that are 

1. It seems strange for this to extend to invalid references in attached routes. Maybe @youngnick has been thinking about this as part of the broader conditions update he's working on?
2. It seems like cross-namespace references that are not allowed by a ReferenceGrant should always receive the RefNotPermitted reason, regardless of if the Secret actually exists.

Reply:
> 2. It seems like cross-namespace references that are not allowed by a ReferenceGrant should always receive the RefNotPermitted reason, regardless of if the Secret actually exists.

This is what I had in mind when I wrote this part. In this case, we should decide which reason has the highest severity:  `RefNotPermitted` or `InvalidCertificateRef`. In my opinion, if no reference is allowed, the reason should be`RefNotPermitted`, no matter if the secret exists or is valid.

Reply:
I agree with what you're suggesting in this comment, I think this PR still needs some work to make that clear in the spec.

Reply:
> 1. It seems strange for this to extend to invalid references in attached routes. Maybe @youngnick has been thinking about this as part of the broader conditions update he's working on?

I just figured out that the previous comment here was copy-paste from the `RouteReasonRefNotPermitted`: https://github.com/kubernetes-sigs/gateway-api/blob/1e037bb54ca1051023c0e9b90f56cdf1ebf0adce/apis/v1beta1/shared_types.go#L247-L251

And it doesn't make any sense here; it misled me. The only reason to have `RefNotPermitted` should be the reference to an ungranted secret. So I'm going to fix it.

Reply:
I addressed both your points, it should be clearer now.